github_slack_bot/utils/slack.py

```diff
@@ -0,0 +1,145 @@
+        logger.debug("Checking if bot '{}' is mentioned in message: '{}'"
+                     .format(bot_user_id, message_stripped))
+        return message_stripped.startswith("<@{}> ".format(bot_user_id))
+
+    # TODO: give user ability to provide a GitHub repo in their message body
+    # TODO: add ability to add assignees,labels, projects, milestones to Issues
+    @staticmethod
+    def parse_bot_message(msg):
+        """Parse a message mentioning our bot
+        :param msg: message sent by user
+        :return: A dict with the issue title and body if they exist,
+        a dict with an error message otherwise
+                Example:
+                {"ok": True, "issue_title": "Issue title here", "issue_body": "Issue body here"}
+                {"ok": False, "error": "Error message goes here"}
+        """
+        msg_regex = "^<@(.+?)> (.*)"
+        matches = re.search(msg_regex, msg)
+        msg_body = matches.group(2).strip()
+        msg_body_split = msg_body.split("__")
+        logger.debug("Split message into {}".format(msg_body_split))
+
+        if len(msg_body_split) == 2:
+            issue_title = msg_body_split[0]
+            issue_body = msg_body_split[1]
+            res = {
+                "ok": True,
+                "issue_title": "{}".format(issue_title),
+                "issue_body": "{}".format(issue_body)
+            }
+        else:
+            res = {
+                "ok": False,
+                "error": "Failed to parse message. Issue title and issue body should be "
+                         "separated by a double underscore (__)"
+            }
+
+        logger.info(res)
+        return res
+
+    def __init__(self, slack_api_token=None, git_client=None):
+        self.git_client = git_client
+        self.slack_api_token = slack_api_token
+        self.bot_user_id = None
+        self.sc = None
+
+        logger.info("Slack bot initialized")
+
+    def set_bot_user_id(self):
+        """Make an auth.test Slack API call to retrieve the bot's user_id.
+        Stores the user_id in the "bot_user_id" instance attribute
+        :return: None
+        """
+        auth_test_res = self.sc.api_call("auth.test")
+        if auth_test_res["ok"]:
+            self.bot_user_id = auth_test_res['user_id']
+            logger.info("Found bot's user_id: {}".format(self.bot_user_id))
+        else:
+            auth_test_error = auth_test_res["error"]
+            logger.error("Failed to retrieve bot's user_id due to: {}"
+                         .format(auth_test_error))
+            raise Exception(EXCEPTION_DEFAULT_MSG)
+
+    # TODO: extend handle_rtm_event() to handle different commands, such as "update-issue", "open-pr"
+    def handle_rtm_event(self, event):
+        """Processes and handles RTM events
+        :param event: Slack RTM event
+        :return: None
+        """
+        if "type" in event and event["type"] == "message" and "subtype" not in event:
```

Comment:
Also, I would prefer to avoid the "staircase" code layout to improve readability by inverting some of the "if" conditions where possible, for example instead of:

```python
if cond1:
  if cond2:
    if cond3:
    else:
```

I'd do:

```python
if not cond1:
  return
if not cond2:
  return
if cond3:
else:
```

Reply:
Cleaned it up a little